lib/translator/common.h

```diff
@@ -43,6 +43,10 @@ struct Allocation {
   uint16_t data_page_count;   // Number of data pages
   uint64_t mdata_addr;        // Start of metadata buffer
   uint16_t mdata_page_count;  // Number of metadata pages
+
+  // Sets [m]data_page_count, calls AllocPages([m]data_page_count),
+  // and returns StatusCode based on whether AllocPages was successful
+  StatusCode SetPages(uint16_t data_page_count, uint16_t mdata_page_count);
```

Comment:
We could implement a safety check to prevent a call to SetPages from overriding page data that has not yet been flushed